source/Modules/Devon4Net.Infrastructure.AWS.CDK/Resources/Handlers/ECS/AwsCdkEcsHandler.cs

```diff
@@ -191,36 +192,63 @@ private Ec2Service CreateEc2Service(Ec2ServiceEntity entity)
                 result.AddPlacementConstraints(new PlacementConstraint[] { PlacementConstraint.DistinctInstances() });
             }
 
-            SetPlacementStrategies(result, entity.PlacementStrategies);
+            SetPlacementStrategy(result, entity.PlacementStrategy, entity.Strategies);
 
             TagHandler.LogTag($"{ApplicationName}{EnvironmentName}{entity.ServiceName}Ec2Service", result);
             return result;
         }
 
-        private void SetPlacementStrategies(Ec2Service entity, List<string> placementStrategies)
+        private void SetPlacementStrategy(Ec2Service entity, string strategy, List<string> strategies)
         {
-            foreach (var strategy in placementStrategies)
+            switch (strategy.ToLower())
             {
-                switch (strategy.ToLower())
-                {
-                    case "binpackmemory":
-                        entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.PackedByMemory() });
-                        break;
-                    case "binpackcpu":
-                        entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.PackedByCpu() });
-                        break;
-                    case "random":
-                        entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.Randomly() });
-                        break;
-                    case "spread_instance":
-                        entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.SpreadAcrossInstances() });
-                        break;
-                    case "spread_az":
-                        entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.SpreadAcross(BuiltInAttributes.AVAILABILITY_ZONE) });
-                        break;
-                    default:
-                        throw new ArgumentException("Please provide one of the following valid Placement Strategies: binpackmemory, binpackcpu, random, spread_az, spread_instance");
-                }
+                case "binpackmemory":
+                    entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.PackedByMemory() });
+                    break;
+                case "binpackcpu":
+                    entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.PackedByCpu() });
+                    break;
+                case "random":
+                    entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.Randomly() });
+                    break;
+                case "spread":
+                    entity.AddPlacementStrategies(new PlacementStrategy[] { PlacementStrategy.SpreadAcrossInstances() });
+                    break;
+                case "custom":
```

Comment:
This could be just one List<string> and keep adding strategies. This approach will get rid of the `custom` strategy and the nested switch. Something like:
```
foreach(string element in strategies)
{
   switch(element.ToLower())
   {
      ...
   }
}
```